plugins/tiddlywiki/browser-sniff/browser.js

```diff
@@ -42,8 +42,10 @@ exports.getInfoTiddlerFields = function() {
 				["is/sailfish","sailfish"],
 				["is/android","android"],
 				["is/windowsphone","windowsphone"],
-				["is/firefoxos","firefoxos"]
+				["is/firefoxos","firefoxos"],
+				["is/mobile","mobile"]
 			];
+		$tw.browser = $tw.utils.extend($tw.browser, bowser.browser);
```

Comment:
`$tw.browser` is set by the startup code, and the core does not expect it to be changed subsequently. What's the idea here?

Reply:
Maybe I can change it to `$tw.utils.extend(bowser.browser, $tw.browser);` ( let `$tw.browser` take priority ), so nothing will be overwritten (though it's not overwritten currently too).

The idea is to add more available info to it, as it already contains some browser info. So some js plugins can use it directly. If you think its an invariant, I can also delete this line, info are already existed in `$:/info/browser` tiddlers, just need some extra step to read them.

Reply:
The trouble is that the properties of `bowser.browser` are not under our direct control, and so subsequent updates to Bowser might overwrite important properties of our own. I'd rather explicitly import the properties that we choose to support.

Reply:
The trouble is that the properties of `bowser.browser` are not under our direct control, and so subsequent updates to Bowser might overwrite important properties of our own. I'd rather explicitly import the properties that we choose to support.

Reply:
Ok, only export mobile, as that is only what I need.